src/Index.js

```diff
@@ -652,12 +652,109 @@ Index.prototype.getSettings = function(callback) {
   var indexObj = this;
   return this.as._jsonRequest({
     method: 'GET',
-    url: '/1/indexes/' + encodeURIComponent(indexObj.indexName) + '/settings',
+    url: '/1/indexes/' + encodeURIComponent(indexObj.indexName) + '/settings?getVersion=2',
```

Comment:
Same for ruby, https://github.com/algolia/algoliasearch-client-ruby/blob/93f7a87a839f43a5d7e06bd9385a837083e34468/lib/algolia/index.rb#L436

Not saying this is good, just searching